### PR TITLE
refactor(scripts): add body class based on page type detected in url

### LINKED ISSUE #241: Add body class based on page type detected in URL
Move page-type-detection and subsequent body-class-adding from client side JS to `helix-markup.yaml`. It would be interesting to see the effect on the experience and lighthouse score if those classes were already set on the server side... currently we do:
- [page type detection here](https://github.com/adobe/theblog/blob/37fbf4f30ce15193118c5db224cefbea811a6e5a/scripts.js#L79-L109) and 
- [add the body class here](https://github.com/adobe/theblog/blob/37fbf4f30ce15193118c5db224cefbea811a6e5a/scripts/common.js#L128)

### PR DESCRIPTION
**Pull request by ramboz**
Use helix markup to inject the class on the html node so it can be read
from the scripts file and we can remove some of the JS logic.

Requires: adobe/helix-pages#385, adobe/helix-pipeline#796

fix #241

### CODE REVIEW

File: scripts.js [hunk 1]
```diff
@@ -91,35 +110,8 @@ window.blog = function() {
...
-          if (segs[1].startsWith(value)) {
-            pageType = value;
-            break;
-          }
-        }
-      }
-    }
-  }
+  const language = LANG[getDetectedLanguage().toUpperCase()] || LANG.EN;
+  const pageType = TYPE[getDetectedPageType().toUpperCase()] || TYPE.HOME;
```
Comment (rofe) [inline scripts.js:114]: I would move the `LANG`/`TYPE` lookup and defaults into the getters for more encapsulation and robustness, so these lines would read:
```suggestion
  const language = getDetectedLanguage();
  const pageType = getDetectedPageType();
```

File: helix-markup.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  home:
+    match: /(|:lang/home)(.*)
+    classnames: home-page
+    type: url
+  author:
+    match: /:lang/author(.*)
+    classnames: author-page
+    type: url
+  post:
+    match: /:lang/(post|drafts/|publish/|\d{4}\/\d{2}\/\d{2}/)(.*)
```
Comment (kptdobe) [inline helix-markup.yaml:13]: Isn't `documentation` missing here ?

Comment (ramboz) [reply helix-markup.yaml:13]: Yes, somehow missed it! Good catch

**Comment by ramboz** [comment]
@rofe I fully agree.

I was never convinced by the "url" match approach as I felt it didn't bubble up the properties high enough in the DOM tree.
This is exactly what we see now. I end up artificially merging on the body element, and then bubbling up in the HTML on the HTML tag.

Feels quite hacky on the pages side, with now huge gain on the blog side… I'm not sure I really prefer the outcome.

**Comment by ramboz** [comment]
@rofe Changed language and type detection logic as requested, and also switch to using `wrap` instead, combined with a fix to the pipeline to properly expose the props on the body/html elements directly. I think this should make the feature more robust

**Comment by ramboz** [comment]
@rofe Still got any concerns on that approach?

**Comment by ramboz** [comment]
> given the fact that it's difficult to determine a true performance gain in this particluar project, I'm still on the fence

👍 Right!

> Having all app logic in javascript sure has its benefits, too.

I guess this depends on what the maintainer of these rules would feel more comfortable with (yaml regex vs. JS).
I'm good either way… don't have any strong feelings on this.

**Comment by rofe** [comment]
Closing for now.

**Pull request closed without merging**